common/persistence/persistence-factory/factory.go

```diff
@@ -79,6 +81,7 @@ type (
 		NewExecutionStore(shardID int) (p.ExecutionStore, error)
 		// NewVisibilityStore returns a new visibility store
 		NewVisibilityStore() (p.VisibilityStore, error)
+		NewQueue(queueType int) (p.Queue, error)
```

Comment:
NewDomainReplicationQueue?

Reply:
This is a general queue which takes a queue type parameter.